core/server/worker/src/main/java/alluxio/worker/block/BlockMetadataManager.java

```diff
@@ -85,13 +85,47 @@ private BlockMetadataManager() {
       if (ServerConfiguration.isSet(PropertyKey.WORKER_EVICTOR_CLASS)) {
         LOG.warn(String.format("Evictor is being emulated. Please use %s instead.",
             PropertyKey.Name.WORKER_BLOCK_ANNOTATOR_CLASS));
-        // Create emulating block iterator.
-        BlockMetadataEvictorView initManagerView = new BlockMetadataEvictorView(this,
-            Collections.<Long>emptySet(), Collections.<Long>emptySet());
-        mBlockIterator = new EmulatingBlockIterator(this,
+        String evictorType = ServerConfiguration.get(PropertyKey.WORKER_EVICTOR_CLASS);
+        switch (evictorType) {
+          case "alluxio.worker.block.evictor.LRUEvictor":
+            LOG.warn("LRUEvictor is deprecated, LRUAnnotator is set instead");
+            ServerConfiguration.set(PropertyKey.WORKER_BLOCK_ANNOTATOR_CLASS,
+                "alluxio.worker.block.annotator.LRUAnnotator");
+            mBlockIterator = new DefaultBlockIterator(this, BlockAnnotator.Factory.create());
+            break;
+          case "alluxio.worker.block.evictor.PartialLRUEvictor":
+            LOG.warn("PartialLRUEvictor is deprecated, LRUAnnotator is set instead");
+            ServerConfiguration.set(PropertyKey.WORKER_BLOCK_ANNOTATOR_CLASS,
+                "alluxio.worker.block.annotator.LRUAnnotator");
+            mBlockIterator = new DefaultBlockIterator(this, BlockAnnotator.Factory.create());
+            break;
+          case "alluxio.worker.block.evictor.LRFUEvictor":
+            LOG.warn("LRFUEvictor is deprecated, LRFUAnnotator is set instead");
+            final double defaultLRFUAttenuationFactor = 2.0;
```

Comment:
The LRFU Evictor related configs like `alluxio.worker.evictor.lrfu.attenuation.factor` and `alluxio.worker.evictor.lrfu.step.factor` have been removed from PropertyKey.java since version 2.3. I think maybe it's better just ignore those two keys, create LRFU annotator with default attenuation factor and step factor of  LRFU annotator. And at the same time, add those configs to RemovedKey.java so that we can track the keys that have been removed and give user warn about the deprecation of it.

Reply:
There are similar ones under block-annotator configuration. 
Please search for:
        `alluxio.worker.block.annotator.lrfu.attenuation.factor`
        `alluxio.worker.block.annotator.lrfu.step.factor`

Reply:
Those two configs are fetched when [LRFUAnnotator is instantiated](https://github.com/Alluxio/alluxio/blob/9e3e9df4bb73b6aeb2c9ef1217d1071e65677cbb/core/server/worker/src/main/java/alluxio/worker/block/annotator/LRFUAnnotator.java#L40), either use default value when they are not specified or use the value user specified explicitly. So I think there is no need to check those two values at `BlockMetadataManager`.

Reply:
Those two configs control how LRFU algorithm works, regardless of their code version.
There's no justification for emulation to mandate its own defaults.

Reply:
The config `alluxio.worker.block.annotator.lrfu.attenuation.factor` and `alluxio.worker.block.annotator.lrfu.step.factor` will be used when user specify Evictor to LRFUEvictor. We will use default value of those two configs if and only if user haven't specified them explicitly. If those two configs are listed in config file, then the specified value will be used.